api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -282,10 +282,10 @@ public DataSet saveDataSet(
 
```

Comment:
Is there no unit test for this class? It would be nice to have a test somewhere which demonstrates the interpolation happening correctly.

(After looking around... looks like there is some testing at the controller level. It may be easiest to add some example of interpolation over there. It's worth noting that we're trying to move more of our unit testing onto services in order to leave controllers as simple as possible... not for this PR, but having no test file for this service class is a bit of a code smell in that regard.)

Reply:
Yeah, this is something Eric has brought up a few times, that we tend to handle our testing at the controller level only. (There's many other cases where we do a bunch of service level tests from the controller tests.)

Might be worth looking into a refactor there to break our tests apart more.